### DISCUSSION TITLE
How to get vertical guides along with showing whitespaces characters like tabs?

### DISCUSSION THREAD

**Discussion by SimonBrandner** [lukas-reineke/indent-blankline.nvim#916] [Q&A] [answered]
In VS Code I can see vertical guides along with tabs:

![image](https://github.com/user-attachments/assets/6a03855c-bea6-4c44-8525-0448b7d85459)

And in Neovim I set `list` and `listchars` to see tabs similarly but it seems to conflict with `ident-blankline`, so this is what I end up with:

![image](https://github.com/user-attachments/assets/65c93f87-80cb-4b96-bcb3-c99b45d7ec54)

Is there any way to work around this? Can I get things looking like in VS Code

**Comment by lukas-reineke** [comment, marked as answer]
`list` and `listchars` do not conflict with IBL. IBL uses the values you have defined in them, see `:help ibl.config.indent.tab_char`.

You cannot render more than one character in the same cell in the terminal. So you cannot have the indentation line, and a different character for the tab at the same time.

As a workaround, you could use a unicode character that combines a line and an arrow, like `󰞘`.

**Reply by SimonBrandner** [reply]
I haven't thought of that and it's a good idea but it's not working much better... 
![image](https://github.com/user-attachments/assets/11cc77b2-cb15-445f-83e8-3c05436304b4)

The tab has a width of 4, wouldn't it somehow be possible to use 1 char for the vertical line and the rest for the tab character?

**Reply by lukas-reineke** [reply]
> The tab has a width of 4, wouldn't it somehow be possible to use 1 char for the vertical line and the rest for the tab character?

You can do that, it's documented in `:help listchars`

**Reply by SimonBrandner** [reply]
![image](https://github.com/user-attachments/assets/011a626a-d8f0-4efa-93d7-41849c238182)

So far this is the best I've managed to do. I've managed to do that by setting `vim.cmd("set listchars=tab:\\▏\\ →\\,space:·")` but that has the arrow character a little too much to the right... Do you have any suggestions?
